plugins/nirmata/api_token.go

```diff
@@ -0,0 +1,89 @@
+package nirmata
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIToken,
+		DocsURL:       sdk.URL("https://nirmata.com/docs/api_token"),               // TODO: Replace with actual URL
+		ManagementURL: sdk.URL("https://console.nirmata.com/user/security/tokens"), // TODO: Replace with actual URL
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to Nirmata.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 116,
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.Email,
+				MarkdownDescription: "Email address registered in Nirmata.",
```

Comment:
Is the email required in the item, since it's not provisioned?

Reply:
But it should be used for this command `nctl login` which prompts for all these fields including email

Reply:
With this shell plugin, the user should no longer have to run `nctl login`. From what I can find in the documentation (`Alternatively, you can use global flags or set the NIRMATA_TOKEN and NIRMATA_URL environment variables.`), this is an alternative to the email-based authentication.

Reply:
Okay in that way makes sense

Reply:
So should should we now make the URL field required and not optional?

Reply:
No, since the docs state that the URL has a default:
```
      --url string     the URL address for Nirmata (env NIRMATA_URL) (default "https://nirmata.io")
```
so not provisioning it would not break the users' workflows.

Reply:
oh okay so should I only import token from the config file?

Reply:
It's okay to also import url from the config file. At provisioning time, if there's nothing to provision for the url, then we can use the default value